File: app/src/main/java/io/mgba/Data/Database/DAOs/CheatDAO.java
package io.mgba.Data.Database.DAOs;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Transaction;
import java.io.File;
import java.util.List;
import io.mgba.Data.Database.Cheat;

@Dao
public interface CheatDAO {

    @Query("SELECT * FROM Cheats WHERE idFK = :key")
    List<Cheat> getGamesCheats(File key);

    @Query("DELETE FROM Cheats WHERE id = :key")
    @Transaction
    void deleteCheat(int key);

    @Insert
    @Transaction
    void insertCheat(Cheat cheat);

}

File: app/src/main/java/io/mgba/data/database/daos/CheatDAO.kt
package io.mgba.data.database.daos

import java.io.File

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Transaction
import io.mgba.data.database.Cheat

@Dao
interface CheatDAO {

    @Query("SELECT * FROM Cheats WHERE idFK = :key")
    fun getGamesCheats(key: File): List<Cheat>

    @Query("DELETE FROM Cheats WHERE id = :key")
    @Transaction
    fun deleteCheat(key: Int)

    @Insert
    @Transaction
    fun insertCheat(cheat: Cheat)

}
